File: sys/frontend/user/mobile/android/app/src/main/kotlin/app/dev/gengoka/domain/model/Answer.kt
package app.dev.gengoka.domain.model

data class Answer(
    val id: String,
    val challengeId: String,
    val userId: String,
    val content: String,
    val score: Int?,
    val aiFeedback: AiFeedback?,
    val likeCount: Int,
    val commentCount: Int,
    val viewCount: Int,
    val createdAt: String
)

data class AnswerWithUser(
    val id: String,
    val challengeId: String,
    val userId: String,
    val content: String,
    val score: Int?,
    val aiFeedback: AiFeedback?,
    val likeCount: Int,
    val commentCount: Int,
    val viewCount: Int,
    val createdAt: String,
    val user: UserSummary,
    val isLiked: Boolean
)

data class AnswerWithDetails(
    val id: String,
    val challengeId: String,
    val userId: String,
    val content: String,
    val score: Int?,
    val aiFeedback: AiFeedback?,
    val likeCount: Int,
    val commentCount: Int,
    val viewCount: Int,
    val createdAt: String,
    val user: UserSummary,
    val challenge: Challenge,
    val isLiked: Boolean
)

data class AiFeedback(
    val score: Int,
    val goodPoints: String,
    val improvement: String,
    val exampleAnswer: String
)

File: sys/frontend/user/web/src/types/answer.ts
import type { Challenge } from './challenge';
import type { UserSummary } from './user';

export interface Answer {
  id: string;
  challenge_id: string;
  user_id: string;
  content: string;
  score: number | null;
  ai_feedback: AiFeedback | null;
  like_count: number;
  comment_count: number;
  view_count: number;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface AiFeedback {
  score: number;
  good_points: string;
  improvement: string;
  example_answer: string;
}

export interface AnswerWithUser extends Answer {
  user: UserSummary;
  is_liked: boolean;
}

export interface AnswerWithDetails extends Answer {
  user: UserSummary;
  challenge: Challenge;
  is_liked: boolean;
}

export interface CreateAnswerRequest {
  content: string;
}
